### PR TITLE
Feature/pok 68 implement calls for info tab

### PR DESCRIPTION
**Pull request by rodrigoschar**
-add call api to update pokemon data in detail
-update pokemon data in database
-add method to get table data by Id

### CODE REVIEW

File: SharedCode/Database/DatabaseManager.cs [hunk 1]
```diff
@@ -96,5 +98,19 @@ public Result<List<T>> GetAllData<T>() where T : new()
                 return Result.Fail<List<T>>("Could not get all data");
             }
         }
+
+        public Result<T> GetDataById<T>(int id, string tableName) where T : new()
+        {
+            try
+            {
+                var query = "SELECT * FROM " + tableName + " WHERE Id LIKE ?";
```
Comment (YawarValeriano) [inline SharedCode/Database/DatabaseManager.cs]: Better use query builder for Sqlite. Refer to the documentation you shared.

**Review by YawarValeriano — CHANGES_REQUESTED** [review]

File: SharedCode/Model/PokemonSpecie.cs [hunk 2]
```diff
@@ -57,6 +57,12 @@ public class GrowthRate
         public string url { get; set; }
     }
 
+    public class Habitat
```
[the remark this answers is not in the document]
Comment (rodrigoschar) [reply SharedCode/Model/PokemonSpecie.cs]: ok I'll checkit

File: SharedCode/Database/DatabaseManager.cs (hunk 1 above)
Comment (rodrigoschar) [reply SharedCode/Database/DatabaseManager.cs]: ok i'll take it a look

**Review by jimibilibob — APPROVED** [review]

File: SharedCode/Database/DatabaseManager.cs (hunk 1 above)
Comment (rodrigoschar) [reply SharedCode/Database/DatabaseManager.cs]: fixed

File: SharedCode/Model/PokemonSpecie.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (rodrigoschar) [reply SharedCode/Model/PokemonSpecie.cs]: fixed

**Review by YawarValeriano — APPROVED** [review]

**Review by adCBueno — APPROVED** [review]

**Pull request merged**